### PR TITLE
use set_montage in edf reader

### PR DESCRIPTION
**Pull request by agramfort**
should be merged and rebased after #6511

### CODE REVIEW

**Comment by massich** [comment]
@agramfort 	this makes it. I would let it like this, for the moment.

**Comment by agramfort** [comment]
ok good enough. thx @massich

File: mne/io/edf/edf.py [hunk 1]
```diff
@@ -280,6 +279,8 @@ def __init__(self, input_fname, montage, eog=None, misc=None,
 
         self.set_annotations(Annotations(onset=onset, duration=duration,
                                          description=desc, orig_time=None))
+        if montage is not None:
+            self.set_montage(montage)
```
Comment (massich) [inline mne/io/edf/edf.py]: For completness, this was never cover and I didn't realize.
The PR modifies RawBDF.__init__ and adds a test for EDF that was already cover.

(I'll update the comment with the commit that amends it)

Comment (massich) [reply mne/io/edf/edf.py]: addressed in #6521

**Pull request merged**